plugins/woocommerce/src/Internal/Utilities/URL.php

```diff
@@ -124,8 +131,11 @@ private function preprocess() {
 	 * without touching the filesystem.
 	 */
 	private function process_path() {
-		$segments          = explode( '/', $this->components['path'] );
-		$this->is_absolute = substr( $this->components['path'], 0, 1 ) === '/';
+		$segments           = explode( '/', $this->components['path'] );
+		$this->is_absolute  = substr( $this->components['path'], 0, 1 ) === '/' || ! empty( $this->components['host'] );
+		$this->is_directory = substr( $this->components['path'], -1, 1 ) === '/' && strlen( $this->components['path'] ) > 1;
```

Comment:
why `strlen( $this->components['path'] ) > 1;`? just `/` is also a directory right?